onedal/cluster/kmeans.py

```diff
@@ -91,9 +91,10 @@ def _check_params_vs_input(
         self, X_table, policy, default_n_init=10, dtype=np.float32
     ):
         # n_clusters
-        if X_table.shape[0] < self.n_clusters:
+        X_row_count = X_table.get_row_count()
+        if X_row_count < self.n_clusters:
```

Comment:
Interface for getting Table's row_count should be:
smth like this `X.row_count`